### ISSUE TITLE
cgeo is unable to retrieve geocache logs after changing date format on geocaching.com

### ISSUE THREAD

**Issue by Lomanic** [cgeo/cgeo#14374]
### Describe your problem!

cgeo is unable to retrieve geocache logs after changing date format on geocaching.com

Log types count is properly displayed, but `The logbook of this cache is empty` is shown instead of logs.

For stored caches, logs are not updated after refreshing.

Also, for newly downloaded event caches, event date is not parsed/stored (shown missing in a stored list).

### How to reproduce?

1. Change the date format at https://www.geocaching.com/account/settings/preferences from MM/dd/yyyy (cgeo historical default value) to yyyy-MM-dd
2. Load a new cache or refresh an already downloaded one

### Actual result after these steps?

Log types count is properly displayed, but `The logbook of this cache is empty` is shown instead of logs.

For stored caches, logs are not updated after refreshing (at least missing new logs).

For newly downloaded event caches, event date is not parsed/stored (shown missing in a stored list)

### Expected result after these steps?

Up-to-date cache logs are found in the Logbook panel

Newly downloaded event caches get their event date properly parsed/stored.

### Reproducible

Yes

### c:geo Version

2023.06.26-NB-eeb5215

### System information

```text
c:geo version: 2023.06.26-NB-eeb5215

Device:
-------
- Device type: Pixel 5 (redfin, google)
- Available processors: 8
- Android version: 13
- Android build: TQ3A.230605.012.2023061402
- Screen resolution: 1080x2072px (393x753dp)
- Pixel density: 2.75
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: unavailable
- Memory:  Available:929.2 MB, Total:7.3 GB, Threshold: 216.0 MB, low:false

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:156
- Set language: en_CA (system default)
- System date format: y-MM-dd
- Time zone: GMT+02:00
- Debug mode active: no
- Log Settings: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
- Last manual backup: Nov 5, 2022, 20:23
- Last auto backup: Jun 15, 19:16
- Routing mode: Walk
- Live map mode: true
- OSM multi-threading: false / threads: 1
- Map: OpenStreetMap.org
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OsmMapSource
  - Atts: © OpenStreetMap DE, map data OpenStreetMap contributors
  - Theme: none

Filters:
-------
- Hide waypoints: -
- LIVE: None ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status))
- OFFLINE: None ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status))

Stored Filters (#0):

Services:
-------
- Geocaching sites enabled:
   - geocaching.com: Logged in (Login OK) / PREMIUM
   - Geocaching.com Adventure Lab
- Geocaching.com date format: MM/dd/yyyy
- Geocaching.com website language: en-US
- Last login error on geocaching.com: communication error (2023-06-13 09-25)
- Last successful login on geocaching.com: 2023-06-26 14-54
- Routing: external / BRouter installed: false
- Installed c:geo plugins:  none

Permissions
-------
- android.permission.ACCESS_COARSE_LOCATION:: granted
- android.permission.ACCESS_FINE_LOCATION:: granted
- android.permission.READ_CONTACTS:: DENIED
- android.permission.READ_EXTERNAL_STORAGE:: DENIED
- android.permission.WRITE_EXTERNAL_STORAGE:: DENIED

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (80.2 GB free) v3 internal isDir(8 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (80.2 GB free) v3 internal isDir(8 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (80.2 GB free) v3 external non-removable isDir(552 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (80.2 GB free) v3 internal isDir(0 entries)
- Map render theme path: 
- Public Folders: #12
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=0, dirs:>=9, totalFileSize:>=0 B, free space: 80.2 GB, files on device: 13511677)
  - OFFLINE_MAPS: /cgeo/maps (Default)[/cgeo/maps[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/maps]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fmaps, Av:true, files:0, dirs:2, totalFileSize:0 B, free space: 80.2 GB, files on device: 13511677)
  - OFFLINE_MAP_THEMES: /cgeo/maps/_themes (Default)[/cgeo/maps/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/maps/_themes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fmaps%2F_themes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 80.2 GB, files on device: 13511677)
  - OFFLINE_MAP_SHADING: /cgeo/maps/_hgt (Default)[/cgeo/maps/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/maps/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fmaps%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 80.2 GB, files on device: 13511677)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 80.2 GB, files on device: 13511677)
  - GPX: /cgeo/gpx (Default)[/cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fgpx, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 80.2 GB, files on device: 13511677)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=8, dirs:>=7, totalFileSize:>=69.1 MB, free space: 80.2 GB, files on device: 13511677)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 80.2 GB, files on device: 13511677)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 80.2 GB, files on device: 13511677)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:7, dirs:1, totalFileSize:81.7 KB, free space: 80.2 GB, files on device: 13511677)
  - ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 80.2 GB, files on device: 13511677)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 80.2 GB, files on device: -1)
- PersistedDocumentUris: #1
  - UNUSED_TRACK: null
- Persisted Uri Permissions: #1
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (Nov 5, 2022, 20:04):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:20.0 MB) on system internal storage

--- End of system information ---
```

### Additional Information

Workaround is to completely exit then relaunch cgeo so it detects the new date format. In the `About: cgeo - System` panel, we can then see the new date format taken into account.

The issue is that cgeo retrieves `serverParameters` only once, at startup/login, [it is then cached](https://github.com/cgeo/cgeo/blob/eeb5215c64f096917987e858c1d98ac9d0145d23/main/src/main/java/cgeo/geocaching/connector/gc/GCLogin.java#L378-L380). If it's not too much work for such an edge case, wouldn't it be possible to force a new retrieval of `/play/serverparameters/params` if cgeo is not able to parse log or event dates?

Relates to #13228

**Comment by ztNFny** [comment]
as you said yourself, c:geo fetches that parameter on login. it's highly unlikely that the date format changes between login and fetching a cache, so imho this should be WONTFIX

**Comment by Lomanic** [comment]
Well, this still happened to me, I was writing quite a different bug report thinking it was the nightly build upgrade I applied from fdroid minutes ago which introduced a regression, until I found by chance in the System information log this line
```
- Geocaching.com date format: MM/dd/yyyy
```
I can only imagine the confusion for a user not knowing how cgeo internals work, not trying to report a bug right away, when even I didn't instantly connect the dots. It is even quite subtle to notice for an already stored geocache as I only initially saw the bug for freshly downloaded caches (obviously missing all logs and showing an incoherent log count), while I also triggered it on at least one already downloaded geocache (missing a single log in that case, while still incrementing the log count above the logs).

Your call to fix this or not of course, at least this is now documented somewhere if anyone else encounters this in the future. Maybe a quick win is to change the `The logbook of this cache is empty` message (in case of date parsing exceptions) to suggest to the user to completely restart the app and retry the geocache download because the date format probably changed since the initial login. And/or show a toast after a date parsing exception (to also handle the case for already downloaded caches) suggesting to restart the app and why, instead of silently swallowing this kind of exception.
